proposals/interface-types/working-notes/Instructions.md

```diff
@@ -0,0 +1,369 @@
+# Interface instructions
+
+This proposal defines new instructions that operate over interface value types. These instructions have the same form as core instructions; they push and pop values on an execution stack and support single pass validation.
+
+The new interface instructions are summarized below.
+
+```
+#interface-instr ::=
+  | bool.lift_iMM
+  | bool.lower_iMM
```

Comment:
booleans SHOULD work out as a use of algebraic types

Reply:
So is the idea that `bool` just gets desugared into a variant, so that `variant.lift`/`lower` Just Work on it?  I suppose this has the minute advantage of avoiding new instructions.  Maybe it makes more sense when we also add `enum` (which is also a special case of a variant where the case payloads are empty).

Reply:
I think `bool` has some value as a builtin type similar to how `s32` has value even though you could just use `i32`. Using `bool` in an interface signals how the interface value is to be interpreted.

It's also a very common type and so we may wish to optimize the use of it more than we could if it was just syntax sugar.

Reply:
I think we all agree there should definitely be a `bool` type; the only question is whether bool values get gets lifted/lowered via new integer-like instructions or using the existing variant lifting/lowering instructions (with the spec desugaring `bool` into a variant).

Reply:
Okay, that makes sense to me. I'll cut `bool` out of this document for now as if it's sugar for a variant, then we'll need `variant` to define `bool`.